### PR TITLE
Get PL course offerings for selected participant type

### PR DESCRIPTION
**Pull request by bethanyaconnor**
Finishes [TEACH-250](https://codedotorg.atlassian.net/browse/TEACH-250) 🥳 PL course offerings are now also returned when getting course offerings for the quick assign, but only if the passed in participantType is not a student.

Getting these courses got a little weird. In particular, we needed to only return courses that were assignable to a section with the provided participant type. However, courses with a participant_audience of student are always assignable to a section, but are in other menus in the quick assign component. This led to a slightly complex filtering, unfortunately.

Preview of the course offerings returned for PL in [this gist](https://gist.github.com/bethanyaconnor/482e3880c83e0b10e5effeb2f6e3f216).

### CODE REVIEW

File: apps/src/templates/sectionsRefresh/CurriculumQuickAssign.jsx [hunk 1]
```diff
@@ -20,7 +21,10 @@ export default function CurriculumQuickAssign({updateSection, sectionCourse}) {
 
   // Retrieve course offerings on mount and convert to JSON
   useEffect(() => {
-    fetch('/course_offerings/quick_assign_course_offerings')
+    const participantType = queryParams('participantType');
```
Comment (hannahbergam) [inline apps/src/templates/sectionsRefresh/CurriculumQuickAssign.jsx:25]: Thanks for updating this!

File: dashboard/app/controllers/course_offerings_controller.rb [hunk 2]
```diff
@@ -25,7 +25,8 @@ def update
   def quick_assign_course_offerings
     return head :forbidden unless current_user
 
-    offerings = QuickAssignHelper.course_offerings(current_user, request.locale)
+    participant_type = params[:participantType] || Curriculum::SharedCourseConstants::PARTICIPANT_AUDIENCE.student
```
Comment (hannahbergam) [inline dashboard/app/controllers/course_offerings_controller.rb]: Do we know when this second case will happen? Or is this a failsafe and we are always expecting a participantType param here?

File: dashboard/lib/quick_assign_helper.rb [hunk 3]
```diff
@@ -13,6 +13,16 @@ def self.course_offerings(user, locale)
     offerings[:high] = group_grade_level_offerings(assignable_high_offerings, user, locale)
     offerings[:hoc] = group_hoc_and_pl_offerings(assignable_hoc_offerings, user, locale)
 
+    unless participant_type == Curriculum::SharedCourseConstants::PARTICIPANT_AUDIENCE.student
+      # Courses with a participant_audience of student are shown in other places
+      # in the quick assign component and should not be included with pl_offerings
```
Comment (hannahbergam) [inline dashboard/lib/quick_assign_helper.rb:18]: Thank you for this

File: dashboard/test/lib/quick_assign_helper.rb [hunk 4]
```diff
@@ -68,9 +124,9 @@ class QuickAssignHelperTest < ActionController::TestCase
     grouped_offerings = QuickAssignHelper.group_grade_level_offerings([course_offering1, course_offering2, course_offering3, course_offering4, course_offering5], teacher, I18n.default_locale)
 
     assert_equal 1, grouped_offerings['Course']['Header 2'].length
-    assert_equal course_offering1.id, grouped_offerings['Course']['Header 1'][0][:id]
+    assert_equal course_offering2.id, grouped_offerings['Course']['Header 1'][0][:id]
```
Comment (hannahbergam) [inline dashboard/test/lib/quick_assign_helper.rb:127]: Just curious about what's going on here

**Review by hannahbergam — APPROVED** [review]
Looks good to me! Left a couple q's/comments, but approving anyway.

File: dashboard/app/controllers/course_offerings_controller.rb (hunk 2 above)
Comment (bethanyaconnor) [reply dashboard/app/controllers/course_offerings_controller.rb]: Good question -- I added it as a failsafe but now I'm wondering if we should reject the request if there isn't a participant type 🤔

File: dashboard/test/lib/quick_assign_helper.rb (hunk 4 above)
Comment (bethanyaconnor) [reply dashboard/test/lib/quick_assign_helper.rb:127]: This was weird. This test started to fail and I'm not sure why. I meant to look into it more then forgot -- I'm a little worried this test isn't being run in Drone for some reason

File: dashboard/app/controllers/course_offerings_controller.rb (hunk 2 above)
Comment (bethanyaconnor) [reply dashboard/app/controllers/course_offerings_controller.rb]: I decided to reject the request. We can always loosen this restriction in the future if needed

**Pull request merged**